### ISSUE TITLE
RecurringTwoAgentWorld poisons rejected transitions with NaN despite being an atomic no-op

### ISSUE THREAD

**Issue by Svector-anu** [SlopDotCash/asi#258]
## What's wrong

\`RecurringTwoAgentWorld._advance_result\` correctly rejects invalid actions and keeps the environment state unchanged, but it then replaces every floating leaf in the returned terminal transition with \`NaN\`:

\`\`\`python
jnp.full_like(value, jnp.nan) if jnp.issubdtype(value.dtype, jnp.floating) else ...
\`\`\`

A caller that records the rejected terminal transition (rather than just checking \`update_rejected\`) therefore receives non-finite observations, rewards, actions, and oracle measurements — poisoning any downstream aggregate metric even though the environment transaction itself is an atomic no-op (state unchanged, action rejected).

Reproduced directly before any fix:

\`\`\`text
applied False terminated True
state_unchanged True
observation_finite False
next_observation_finite False
reward_finite False
next_observation [[nan nan nan nan nan nan]
 [nan nan nan nan nan nan]]
\`\`\`

## Reachability

\`RecurringTwoAgentWorld\` is exported from \`alberta_framework.streams\` and included in that subpackage's \`__all__\`. Its public \`step\`/\`step_result\` and \`step_with_partner\`/\`step_with_partner_result\` methods both delegate to the affected transactional path (\`_advance_result\`) with caller-supplied actions — non-finite/out-of-range real inputs reach the corrected rejection behavior through public entry points. It is not re-exported from the top-level \`alberta_framework\` namespace.

## Fix

Rejected transition leaves now use finite zeros for inexact (floating/complex) arrays instead of \`NaN\`. Boolean and integer rejection sentinels retain their existing behavior (\`False\` / \`-1\` respectively). The committed state remains unchanged and the transition remains correctly marked terminal with zero discount — only the poisoned NaN leaves are replaced with a finite, well-defined sentinel.

PR incoming with a regression test using a non-finite action from a valid state, asserting rejection, terminal status, full finiteness, and an exact no-op state, plus a mutation-resistance transcript.

**Comment by lalalune** [comment]
Maintainer review correction: finite out-of-range actions are not rejected by this API; they are clipped to `[-1, 1]` and applied. The reachable public rejection reproducer is the non-finite action shown here (plus invalid state/lifetime cases). This narrows the reachability wording but does not change the defect or repair: every actually rejected transition now has finite floating payloads, terminal status, zero discount, and exact state rollback.
